src/go/rpk/pkg/cli/topic/unmount.go

```diff
@@ -0,0 +1,130 @@
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newUnmountCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var namespace string
+	var timeout int
+
+	cmd := &cobra.Command{
+		Use:   "Unmount [TOPIC]",
+		Short: "Unmount a topic from the Redpanda Cluster",
+		Long:  helpUnmount,
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, topics []string) {
+			pf, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "rpk unable to load config: %v", err)
+			config.CheckExitCloudAdmin(pf)
+			adm, err := adminapi.NewClient(fs, pf)
+			out.MaybeDie(err, "unable to initialize admin client: %v", err)
+
+			resp, err := adm.AddOutboundMigration(cmd.Context(), rpadmin.OutboundMigration{
+				MigrationType: "outbound",
+				Topics: []rpadmin.Topic{
+					{
+						Topic:     topics[0],
+						Namespace: namespace,
+					},
+				},
+				ConsumerGroups: []string{}, // not implemented yet
+			})
+			out.MaybeDie(err, "unable to add outbound migration: %v", err)
+
+			tmt := time.Duration(timeout * int(time.Minute/time.Nanosecond))
+			if errP := checkMigrationActionAndAdvanceState(cmd.Context(), resp.ID, adm, rpadmin.PrepareMigrationAction, "prepared", tmt); err != nil {
+				out.MaybeDie(errP, "unable to prepare migration", errP)
+			}
+			if errP := checkMigrationActionAndAdvanceState(cmd.Context(), resp.ID, adm, rpadmin.ExecuteMigrationAction, "executed", tmt); err != nil {
+				out.MaybeDie(errP, "unable to execute migration", errP)
+			}
+			if errP := checkMigrationActionAndAdvanceState(cmd.Context(), resp.ID, adm, rpadmin.FinishMigrationAction, "finished", tmt); err != nil {
+				out.MaybeDie(errP, "unable to finish migration", errP)
+			}
+		},
+	}
+	cmd.Flags().StringVarP(&namespace, "namespace", "n", "", "Namespace for the topic (optional, defaults to 'kafka')")
+	cmd.Flags().IntVarP(&timeout, "timeout", "", 5, "Timeout in minutes for the unmount process to finish (default: 5)")
+	return cmd
+}
+
+type migrationClient interface {
+	ExecuteMigration(ctx context.Context, id int, action rpadmin.MigrationAction) error
+	GetMigration(ctx context.Context, id int) (rpadmin.MigrationState, error)
+}
+
+func checkMigrationActionAndAdvanceState(ctx context.Context, id int, adm migrationClient, doingAction rpadmin.MigrationAction, desiredState string, timeout time.Duration) error {
+	if err := adm.ExecuteMigration(ctx, id, doingAction); err != nil {
+		return fmt.Errorf("unable to execute migration: %w", err)
+	}
+
+	timeoutChan := time.After(timeout)
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-timeoutChan:
+			return fmt.Errorf("timeoutChan reached: migration did not reach desired state %s and timed out", desiredState)
+		case <-ticker.C:
+			m, err := adm.GetMigration(ctx, id)
+			if err != nil {
+				return fmt.Errorf("unable to get migration state: %s", err)
+			}
+			if m.State == desiredState {
```

Comment:
wonder if the desiredState should also be constants as you do in `rpadmin.FinishMigrationAction`

Reply:
I agree, the only reason I didn't do it is that it would make a lot more sense to put it in common-go